modules/vmfcore/test/test_statistics.cpp

```diff
@@ -0,0 +1,837 @@
+        EXPECT_NO_THROW( op = vmf::StatOpFactory::create( name ));
+        ASSERT_NE( op, nullptr );
+        EXPECT_NO_THROW( str = op->name() );
+        ASSERT_EQ( str, name );
+        delete op; op = nullptr;
+    }
+
+    class UserOp: public vmf::StatOpBase
+    {
+    public:
+        UserOp()
+            {}
+        virtual ~UserOp()
+            {}
+
+    public:
+        virtual std::string name() const
+            { return userOpName; }
+        virtual void reset()
+            { m_value = vmf::Variant(); }
+        virtual bool handle( vmf::StatAction::Type action, const vmf::Variant& fieldValue )
+            {
+                switch( action )
+                {
+                case vmf::StatAction::Add:
+                    m_value = fieldValue;
+                    break;
+                case vmf::StatAction::Remove:
+                    return false;
+                }
+                return true;
+            }
+        virtual vmf::Variant value() const
+            { return m_value; }
+
+    private:
+        vmf::Variant m_value;
+
+    public:
+        static const std::string userOpName;
+        static vmf::StatOpBase* createInstance()
+            { return new UserOp(); }
+        static vmf::StatOpBase* createInstance2()
+            { return new UserOp(); }
+    };
+};
+
+/*static*/ const std::string TestStatOperations::UserOp::userOpName = "***TestStatOperations::UserOp::userOpName***";
+
+TEST_F( TestStatOperations, BuiltinMin )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Min ),
+                 InputInt | InputReal | OutputSame | ResetEmpty | CanAdd );
+}
+
+TEST_F( TestStatOperations, BuiltinMax )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Max ),
+                 InputInt | InputReal | OutputSame | ResetEmpty | CanAdd );
+}
+
+TEST_F( TestStatOperations, BuiltinAverage )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Average ),
+                 InputInt | InputReal | OutputReal | ResetEmpty | CanAdd | CanRemove );
+}
+
+TEST_F( TestStatOperations, BuiltinCount )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Count ),
+                 InputAny | OutputInt | ResetInt | CanAdd | CanRemove );
+}
+
+TEST_F( TestStatOperations, BuiltinSum )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Sum ),
+                 InputInt | InputReal | OutputSame | ResetEmpty | CanAdd | CanRemove );
+}
+
+TEST_F( TestStatOperations, BuiltinLast )
+{
+    testBuiltin( vmf::StatOpFactory::builtinName( vmf::StatOpFactory::BuiltinOp::Last ),
+                 InputAny | OutputSame | ResetEmpty | CanAdd );
+}
+
+TEST_F( TestStatOperations, UserOperation )
+{
+    testUserOperation();
+}
+
+class TestStatistics : public ::testing::TestWithParam< vmf::StatUpdateMode::Type >
+{
+protected:
+    void SetUp()
+    {
+        vmf::initialize();
```

Comment:
Actually, it tests the factory: registration & creation mechanics and some exceptions thrown/not thrown there. But to do so, it would be need to register & create some dumb operation... Perhaps it should be renamed appropriately